src/overlaybd/tar/erofs/liberofs.cpp

```diff
@@ -580,20 +580,29 @@ static int erofs_init_tar(struct erofs_tarfile *erofstar,
 static int erofs_write_map_file(photon::fs::IFile *fout, uint64_t blksz, FILE *fp)
 {
     uint64_t blkaddr, toff;
-    uint32_t nblocks;
+    uint32_t nblocks, zeroedlen;
+    char line[blksz];
+    int cnt;
 
     if (fp == NULL) {
        LOG_ERROR("unable to get upper.map, ignored");
        return -1;
     }
     rewind(fp);
-    while (fscanf(fp, "%" PRIx64" %x %" PRIx64 "\n", &blkaddr, &nblocks, &toff)
-           >= 3)
-    {
+
+    while (fgets(line, sizeof(line), fp)) {
         LSMT::RemoteMapping lba;
+
+        cnt = sscanf(line, "%" PRIx64" %x %" PRIx64 "%08u\n", &blkaddr, &nblocks, &toff, &zeroedlen);
+        if (!(cnt == 3 || cnt == 4))
+            LOG_ERRNO_RETURN(0, -1, "fail to read block list");
+
         lba.offset = blkaddr * blksz;
         lba.count = nblocks * blksz;
```

Comment:
lba.count = nblocks * blksz - round_down_blk(lba.count - zeroedlen); ?

Are you sure that is round_up_blk?